src/client/routes/Spigot/index.tsx

```diff
@@ -71,7 +72,9 @@ export const Spigot = () => {
     // TODO: implement getSpigot gql query and replace dispatch to getLinePage
     // dispatch(CollateralActions.getSpigotPage({ id: spigotAddress }));
     if (spigotAddress && !selectedSpigot) {
-      dispatch(LinesActions.getLinePage({ id: lineAddress }));
+      dispatch(TokensActions.getTokens())
```

Comment:
If we reuse it across components should abstract it into the Router or some HoC or create a hook that gets used everywhere. Should also be checking against state if the only reason we need to call is if state isnt populated already.

Reply:
did not understand what the benefit / desired change was for using a higher order component or hook given the simplicity of the logic. checks against state added to only dispatch `getTokens()` when `tokensMap` is undefined